wpimath/src/main/java/edu/wpi/first/math/controller/SimpleMotorFeedforward.java

```diff
@@ -59,40 +110,44 @@ public SimpleMotorFeedforward(double ks, double kv) {
    * @param acceleration The acceleration setpoint.
    * @return The computed feedforward.
    */
+  @SuppressWarnings("removal")
+  @Deprecated(forRemoval = true, since = "2025")
   public double calculate(double velocity, double acceleration) {
     return ks * Math.signum(velocity) + kv * velocity + ka * acceleration;
   }
 
   /**
-   * Calculates the feedforward from the gains and setpoints.
+   * Calculates the feedforward from the gains and velocity setpoint (acceleration is assumed to be
+   * zero).
    *
-   * @param currentVelocity The current velocity setpoint.
-   * @param nextVelocity The next velocity setpoint.
-   * @param dtSeconds Time between velocity setpoints in seconds.
+   * @param velocity The velocity setpoint.
    * @return The computed feedforward.
    */
-  public double calculate(double currentVelocity, double nextVelocity, double dtSeconds) {
-    var plant = LinearSystemId.identifyVelocitySystem(this.kv, this.ka);
-    var feedforward = new LinearPlantInversionFeedforward<>(plant, dtSeconds);
-
-    var r = MatBuilder.fill(Nat.N1(), Nat.N1(), currentVelocity);
-    var nextR = MatBuilder.fill(Nat.N1(), Nat.N1(), nextVelocity);
-
-    return ks * Math.signum(currentVelocity) + feedforward.calculate(r, nextR).get(0, 0);
+  @SuppressWarnings("removal")
+  @Deprecated(forRemoval = true, since = "2025")
+  public double calculate(double velocity) {
+    return calculate(velocity, 0);
   }
 
-  // Rearranging the main equation from the calculate() method yields the
-  // formulas for the methods below:
-
   /**
-   * Calculates the feedforward from the gains and velocity setpoint (acceleration is assumed to be
-   * zero).
+   * Calculates the feedforward from the gains and setpoints assuming discrete control.
    *
-   * @param velocity The velocity setpoint.
+   * @param <U> The velocity parameter either as distance or angle.
+   * @param currentVelocity The current velocity setpoint.
+   * @param nextVelocity The next velocity setpoint.
    * @return The computed feedforward.
    */
-  public double calculate(double velocity) {
-    return calculate(velocity, 0);
+  public <U extends Unit<U>> double calculate(
+      Measure<Velocity<U>> currentVelocity, Measure<Velocity<U>> nextVelocity) {
+    if (ka == 0.0) {
+      return ks * Math.signum(nextVelocity.magnitude()) + kv * nextVelocity.magnitude();
```

Comment:
Here's a derivation we can include so people don't try to change this later on accident.
```
Given the following discrete feedforward model

  uₖ = B_d⁺(rₖ₊₁ − A_d rₖ)

where

  A_d = eᴬᵀ
  B_d = A⁻¹(eᴬᵀ - I)B
  A = −kᵥ/kₐ
  B = 1/kₐ

We want the feedforward model when kₐ = 0.

Simplify A.

  A = −kᵥ/kₐ

As kₐ approaches zero, A approaches -∞.

  A = −∞

Simplify A_d.

  A_d = eᴬᵀ
  A_d = exp(−∞)
  A_d = 0

Simplify B_d.

  B_d = A⁻¹(eᴬᵀ - I)B
  B_d = A⁻¹((0) - I)B
  B_d = A⁻¹(-I)B
  B_d = -A⁻¹B
  B_d = -(−kᵥ/kₐ)⁻¹(1/kₐ)
  B_d = (kᵥ/kₐ)⁻¹(1/kₐ)
  B_d = kₐ/kᵥ(1/kₐ)
  B_d = 1/kᵥ

Substitute these into the feedforward equation.

  uₖ = B_d⁺(rₖ₊₁ − A_d rₖ)
  uₖ = (1/kᵥ)⁺(rₖ₊₁ − (0) rₖ)
  uₖ = kᵥrₖ₊₁
```

Reply:
added to the java version.  
What am I doing wrong in the SimpleMotorFeedforward.h file at line 97.  feedforward isn't accepting r and nextR

Reply:
I think it has something to do with the type templete in the plant inversion class, but I also think it could be I've created r and nextR incorrectly?